### ISSUE TITLE
Add some way to write the Infra list/etc literals in ASCII

### ISSUE THREAD

**Issue by tabatkins** [tabatkins/bikeshed#1526]
You write out an Infra list literal like « "a", "b", "c", "a" », and do map iteration like key → value. These non-ASCII characters are nice for grabbing attention and being unambiguous in text, but they're annoying to type - I tend to just go copy-paste from the Infra spec.

We should come up with an ASCII way to write these literals in your source, that'll turn into the nice unicode characters in the output.

/cc @marcoscaceres to match up in syntax.

**Comment by tabatkins** [comment]
Things to address:

* [ ] `« "a", "b", "c", "a" »` (list literal)
* [ ] `«[ "a" → x, "b" → y ]»` (map literal)
* [ ] `For each key → value of map` (map for-each)

And the ASCII ones, which are typeable but maybe worth including as well: 
* [ ] `` `foo` `` (bytestring literal)
* [ ] `(200, "OK")` (tuple literal)
* [ ] `200/"OK"` (pair literal)

I'm thinking of doing some simple fake elements, like I do for a some other things like `<css>`.  So `<infra-list>foo, bar, baz</infra-list>`, and translate it into `<span>«foo, bar, baz»</span>`. Similarly, `<infra-map>foo -> bar, baz -> qux</infra-map>` (ascii arrows) would become `<span>«[ foo → bar, baz → qux ]»</span>`. And the same for the ASCII types, I guess? `<infra-bytes>`, etc.

Unclear what to do about for-each, tho. Maybe `<infra-foreach>key -> value of map</infra-foreach>`, changing the `->` to an arrow and adding the "For each" opening?

**Comment by domenic** [comment]
I would find those to be too verbose to use, personally. Especially when reading them, although writing them would also be a pain.

**Comment by tabatkins** [comment]
Hm. Any suggestions for better? I'd prefer avoiding use of `<` in an ASCII-ish shorthand.

**Comment by domenic** [comment]
Yeah, I mean, `<<` is already taken. It looks like `[[` is not, so, I dunno... `[[ "a", "b", "c" ]]`, `[[ "a" -> x, "b" -> y ]]`, and then... `-->` for the foreach case? If they're behind an opt-in this might not collide.

**Comment by tabatkins** [comment]
Hehe, [[]] is biblio links already
